airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -300,21 +320,60 @@ public static void main(final String[] args) throws IOException, InterruptedExce
     final Optional<SecretPersistence> ephemeralSecretPersistence = SecretPersistence.getEphemeral(configs);
     final ConfigRepository configRepository = new ConfigRepository(configPersistence, secretsHydrator, secretPersistence, ephemeralSecretPersistence);
 
+    final Database jobDatabase = new JobsDatabaseInstance(
+        configs.getDatabaseUser(),
+        configs.getDatabasePassword(),
+        configs.getDatabaseUrl())
+            .getInitialized();
+
+    final JobPersistence jobPersistence = new DefaultJobPersistence(jobDatabase);
```

Comment:
side comment - this method is getting hard to grok..

Reply:
Yeah I agree, I wanna try https://github.com/airbytehq/airbyte/issues/8598 to make it easier to understand but it's a fairly big issue. I don't think that it will be something that we can easily do incrementally.

Reply:
agree. we can also have a separate issue for this. will leave it up to you.